static/js/protein/data_processing_utils.ts

```diff
@@ -0,0 +1,400 @@
+    }
+    for (const node of neighbour.nodes) {
+      let variant = null;
+      for (const n of node.neighbors) {
+        if (n.property !== "geneID") {
+          continue;
+        }
+        for (const n1 of n.nodes) {
+          for (const n2 of n1.neighbors) {
+            if (n2.property === "geneticVariantFunctionalCategory") {
+              //check for empty values
+              if (_.isEmpty(n2.nodes[0].value)) {
+                continue;
+              }
+              const count = 1;
+              variant = n2.nodes[0].value;
+              result.push({ name: variant, value: count });
+            }
+          }
+        }
+      }
+    }
+    const returnData = [];
+    // combine all the variant counts based on similar categories
+    result.forEach(function (element) {
+      if (!this[element.name]) {
+        this[element.name] = { name: element.name, value: 0 };
+        returnData.push(this[element.name]);
+      }
+      this[element.name].value += element.value;
+    }, Object.create(null));
+    return returnData;
+  }
+  return [];
+}
+
+export function getVarSigAssoc(data: GraphNodes): ProteinNumData[] {
+  // Variant Gene Associations
+  if (!data) {
+    return [];
+  }
+  const result: ProteinNumData[] = [];
+  // check for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "geneID") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let variant = null;
+      for (const n of node.neighbors) {
+        if (n.property !== "geneID") {
+          continue;
+        }
+        for (const n1 of n.nodes) {
+          for (const n2 of n1.neighbors) {
+            if (n2.property === "clinicalSignificance") {
+              if (_.isEmpty(n2.nodes[0].value)) {
+                continue;
+              }
+              let count = 0;
+              variant = n2.nodes[0].value;
+              count = 1;
+              result.push({ name: variant, value: count });
+            }
+          }
+        }
+      }
+    }
+    const returnData = [];
+    result.forEach(function (element) {
+      if (!this[element.name]) {
+        this[element.name] = { name: element.name, value: 0 };
+        returnData.push(this[element.name]);
+      }
+      this[element.name].value += element.value;
+    }, Object.create(null));
+    return returnData;
+  }
+  return [];
+}
+
+export function getChemicalGeneAssoc(data: GraphNodes): ProteinNumData[] {
+  // Chem Gene Associations
+  if (!data) {
+    return [];
+  }
+  const result: { name: string; value: number }[] = [];
+  // check for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "geneID") {
+      continue;
+    }
+    if (_.isEmpty(neighbour.nodes)) {
+      return [];
```

Comment:
I don't think you want to return an empty array here?

Reply:
Fixed it !